### DISCUSSION TITLE
Cannot register PnP.Powershell - Authentication Failed

### DISCUSSION THREAD

**Discussion by NateMetz** [pnp/powershell#2692] [Q&A]
Hi,
I have been trying to register PnP.Powershell in my tenancy to use with SharePoint online. I am a global admin and can connect and use Connect-SPOService with no issues. When I try to run Register-PnPManagementShellAccess I get the following error:

"You did not consent for the PnP Management Shell Application for use by PnP PowerShell. Feel free to close this browser window.
Error details: error invalid_client error_description:AADSTS650052: The app is trying to access a service '...'(Dataverse) that your organization '...' lacks a service principal for. Contact your IT Admin to review the configuration of your service subscriptions or consent to the application in order to create the required service principal. Trace ID: 5fa6c6a2-4dae-4313-8740-a18e31f65802 Correlation ID: 42c25333-7cf3-4e6e-b51b-aa92ed59a3b8

The usual screen listing all permissions required does not appear, and jumps to the error.

I have also tried to use the -ShowConsentUrl option to no avail.
I have tried this in my personal Dev environment with no issues.
I have disabled MFA for my account

Any assistance would be greatly appreciated.
Cheers,
Nathan

**Comment by marcoszorzi** [comment]
Hi @NateMetz were you able to figure this one out? I'm facing the same issue with a customer.

**Comment by AndersRask** [comment]
Dataverse... Thats Power Platform. And the error looks like a missing license. 
Not exactly sure why that would be needed, but as a test maybe check licenses for the admin user in AAD and try to assign it something with Flow/Power Platform like an E5 or similar.
